### PR TITLE
[Aruba] Add parsing of CoPP events

### PR DESCRIPTION
**Pull request by dwhyrock**
Issue: https://github.com/elastic/integrations/issues/11542

This PR adds in logs for HPE Aruba CoPP log types.

The logs were hand-crafted based on HPE's page: https://www.arubanetworks.com/techdocs/AOS-CX/10.07/HTML/5200-8214/Content/events/COPP.htm

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log [hunk 1]
```diff
@@ -1,3 +1,16 @@
+2024-06-18T12:45:38.182641-05:00 8360-Primaire hpe-copp[1233]: Event|1501|LOG_ERR|COPP|-|CoPP initialization failed
```
Comment (qcorporation) [inline packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:1]: It's nice to have, but I believe we've decided to omit pure log lines that do not require parsing. Can keep it in if you'd like

Comment (gogochan) [inline packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:1]: @qcorporation suggested that we don't include a log that doesn't have any parameter in it. I think we can eliminate 1501 through 1505.

File: packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+2024-06-18T12:46:38.182641-05:00 8360-Primaire hpe-copp[1233]: Event|1502|LOG_ERR|COPP|-|COPP initialization successful
```
Comment (qcorporation) [inline packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:2]: Severity is Information here if you decide to keep this log line

**Review by gogochan — APPROVED** [review]
Looks Good to me, with a minor suggestion.

**Review by qcorporation — COMMENTED** [review]
In general very good - just one small issue

File: packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log (hunk 1 above)
Comment (taylor-swanson) [reply packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:1]: I think it's always good to include test cases, even if no additional processing occurs. That may not always be the case in some situations.

Comment (taylor-swanson) [reply packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:1]: I disagree, I think we should continue to include these, if nothing else but for regression testing. There's no harm in keeping these here.

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -37,8 +37,66 @@ processors:
         field: event.kind
         tag: lowercase_event_kind
         ignore_missing: true
-    
-    # ECMP Events
+
+    # CoPP Events (15xx)
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:41]: Could we include a link to the docs page for each log type as well? That has proved very helpful in other integrations (panw comes to mind). I know the README has it, but it's nice when it's right next to the code.

```suggestion
    # CoPP Events (15xx)
    # https://www.arubanetworks.com/techdocs/AOS-CX/10.07/HTML/5200-8214/Content/events/COPP.htm
```

File: packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -222,6 +280,14 @@ processors:
         patterns:
             - "%{DATA:aruba.dns.type} event for VRF %{DATA:aruba.dns.vrf_name}$"
 
+    # Convert aruba.slot from a string to a long
+    # Dissect processor sets everything to a string type
+    - convert:
```
Comment (taylor-swanson) [inline packages/hpe_aruba_cx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:285]:
```suggestion
    - convert:
        tag: convert_aruba_slot
```

File: packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log (hunk 1 above)
Comment (dwhyrock) [reply packages/hpe_aruba_cx/data_stream/log/_dev/test/pipeline/test-aruba-fabricated.log:1]: Yea, I think I'll leave these in.  Guards against future bugs, as Taylor mentioned.

**Review by taylor-swanson — APPROVED** [review]
LGTM

**Pull request merged**